packages/fiori/src/DynamicSideContent.js

```diff
@@ -0,0 +1,546 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import { getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import ResizeHandler from "@ui5/webcomponents-base/dist/delegate/ResizeHandler.js";
+import SideContentPosition from "./types/SideContentPosition.js";
+import SideContentVisibility from "./types/SideContentVisibility.js";
+import SideContentFallDown from "./types/SideContentFallDown.js";
+import DynamicSideContentTemplate from "./generated/templates/DynamicSideContentTemplate.lit.js";
+
+// Styles
+import DynamicSideContentCss from "./generated/themes/DynamicSideContent.css.js";
+
+// Texts
+import {
+	DSC_SIDE_ARIA_LABEL,
+} from "./generated/i18n/i18n-defaults.js";
+
+/**
+ * @public
+ */
+const metadata = {
+	tag: "ui5-dynamic-side-content",
+	managedSlots: true,
+	properties: /** @lends sap.ui.webcomponents.main.DynamicSideContent.prototype */ {
+
+		/**
+		 * Determines whether to hide the main content.
+		 *
+		 * @type {boolean}
+		 * @public
+		 *
+		 */
+		 hideMainContent: {
+			type: Boolean,
+		},
+
+		/**
+		 * Determines whether to hide the side content.
+		 *
+		 * <b>Note:</b> If both <code>hideSideContent</code> and <code>hideMainContent</code>
+		 * properties are set to <code>false</code>, use the <code>toggleContents</code> method
+		 * for showing the side content on phone.
+		 *
+		 * @type {boolean}
+		 * @public
+		 *
+		 */
+		hideSideContent: {
+			type: Boolean,
+		},
+
+		/**
+		 * Determines whether the side content is on the left or on the right side of the main content.
+		 *
+		 * @type {SideContentPosition}
+		 * @defaultvalue "End"
```

Comment:
you have to explain all the available options here and for sideContentPosition, sideContentVisibility

you can refer to some existing property descriptions to make it consistent